packages/expo/scripts/aggregate_xcprivacy/aggregate_xcprivacy.rb

```diff
@@ -0,0 +1,91 @@
+require_relative './plist/lib/plist'
+
+def get_used_required_reason_apis(installer)
+  # A dictionary with keys of type string (NSPrivacyAccessedAPIType) and values of type string[] (NSPrivacyAccessedAPITypeReasons[])
+  used_apis = {}
+  puts "Reading xcprivacy files to aggregate all used Required Reason APIs."
+  installer.pod_targets.each do |pod_target|
+    pod_target.file_accessors.each do |file_accessor|
+      file_accessor.resource_bundles.each do |bundle_name, bundle_files|
+        bundle_files.each do |file_path|
+          # This needs to be named like that due to apple requirements
+          if File.basename(file_path) == 'PrivacyInfo.xcprivacy'
+            content = Plist.parse_xml(file_path)
+            accessed_api_types = content["NSPrivacyAccessedAPITypes"]
+            accessed_api_types.each do |accessed_api|
+              api_type = accessed_api["NSPrivacyAccessedAPIType"]
+              reasons = accessed_api["NSPrivacyAccessedAPITypeReasons"]
+              used_apis[api_type] ||= []
+              used_apis[api_type] += reasons
+            end
+          end
+        end
+      end
+    end
+  end
+  return used_apis
+end
+
+def get_privacyinfo_file_path(target)
+  # We find a file we know exists in the project to get the path to the main group directory
+  app_delegate_path = target.source_build_phase.files_references.find { |file_ref| file_ref.name == "AppDelegate.mm" }
+  destination_path = "#{File.dirname(app_delegate_path.real_path)}/PrivacyInfo.xcprivacy"
+  return destination_path
+end
+
+def read_privacyinfo_file(file_path)
+  # Maybe add missing default NSPrivacyTracking, NSPrivacyTrackingDomains, NSPrivacyCollectedDataTypes, but this works without those keys
+  source_data = nil
+  # Try to read an exisitng PrivacyInfo.xcprivacy file
+  begin
+    source_data = Plist.parse_xml(file_path)
+    puts "Appending aggregated reasons to existing PrivacyInfo.xcprivacy file."
+  rescue => e
+    puts "No existing PrivacyInfo.xcprivacy file found, creating a new one."
+  end
+  return source_data
+end
+
+def ensure_reference(file_path, target)
+  reference_exists = target.resources_build_phase.files_references.any? { |file_ref| file_ref.name == "PrivacyInfo.xcprivacy" }
+  unless reference_exists
+    file_ref = project.new_file(file_path)
+    build_file = resources_build_phase.add_file_reference(file_ref, true)
+  end
+end
+
+def aggregate_xcprivacy!(installer)
+  # Get all required reason APIs defined in current pods
+  required_reason_apis = get_used_required_reason_apis(installer)
+
+  # Get a reference to the Xcode project
+  project = installer.aggregate_targets.first.user_project
```

Comment:
The default iOS app we generate has 2 targets: `AppName` and `AppNameTests`.
A better approximation might be:
1. get all the names
2. filter out the names that include `Test` in the name
3. If only 1 target remains, adds the privacy manifest to it
4. If there are no targets or more than one target (for example an app called Testify will ends up with Testify and TestifyTests targets), pick the first.

We can also provide a level of customization: we can emit an error if we can't find the proper target and, in the error, tell the user how to modify the Podfile to provide the TargetName to which the privacy manifest must be associated.

Another possible approach, is to associate it to all the targets (generating more PrivacyManifests): most common apps have 1 real target and 1 or 2 test targets (Unit tests and UITests). Test target are not required to have a privacy manifest, but adding it will not hurt.

Reply:
Afaik we also need to ignore non-app targets, like watchOS extensions, widgets, etc.

Reply:
If we somehow can get the "native target", we could use this: https://github.com/CocoaPods/Xcodeproj/blob/master/lib/xcodeproj/project/object/native_target.rb#L471-L473 and https://github.com/CocoaPods/Xcodeproj/blob/master/lib/xcodeproj/constants.rb#L151